### PR TITLE
Handle REST status codes

### PR DESCRIPTION
**Pull request by isinyaaa**
Closes: https://github.com/opendatahub-io/model-registry-bf4-kf/issues/305

## Description
This handles REST status codes coming from the server core layer using enums and error unwrapping as [introduced in GO 1.13](https://go.dev/blog/go1.13-errors).

## How Has This Been Tested?
Tested with httpie and go tests.

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: internal/server/openapi/error.go [hunk 1]
```diff
@@ -15,10 +15,8 @@ import (
 	"net/http"
 )
 
-var (
-	// ErrTypeAssertionError is thrown when type an interface does not match the asserted type
-	ErrTypeAssertionError = errors.New("unable to assert type")
-)
+// ErrTypeAssertionError is thrown when type an interface does not match the asserted type
+var ErrTypeAssertionError = errors.New("unable to assert type")
```
Comment (tarilabs) [inline internal/server/openapi/error.go:19]: I'm a bit perplexed why this generated file has been modified, and allegedly is not re-generated throwing error during GHA for uncommitted modifications; can you elaborate, please?

File: internal/server/openapi/api_model_registry_service_service.go [hunk 2]
```diff
@@ -67,12 +70,17 @@ func (s *ModelRegistryServiceAPIService) CreateInferenceServiceServe(ctx context
 
 	result, err := s.coreApi.UpsertServeModel(entity, &inferenceserviceId)
 	if err != nil {
+		if errors.Is(err, api.ErrBadRequest) {
+			return Response(400, model.Error{Message: err.Error()}), nil
+		}
+		if errors.Is(err, api.ErrNotFound) {
+			return Response(404, model.Error{Message: err.Error()}), nil
+		}
```
Comment (tarilabs) [inline internal/server/openapi/api_model_registry_service_service.go:78]: eventually, there can be a mapping function as discussed which for known-errors will return a `Response` if mapped or nil, so we can refactor similar code here.

**Review by tarilabs — COMMENTED** [review]
Thank you @isinyaaa !

I prefer for this PR to consider changing return error codes also for failed conversions like `s.converter ... ` in `api_model_registry_service_service.go` instead of 500 to return 400 for bad request, as discussed.

I'm not sure I understood `internal/server/openapi/error.go` modification, per comment below.

The mapping function, can be done in a subsequent PR.

File: internal/server/openapi/error.go (hunk 1 above)
Comment (isinyaaa) [reply internal/server/openapi/error.go:19]: I had previously tried to introduce the new error codes in there, as discussed on last Wednesday's call. Missed the autoformatter change, will remove it.

**Review by tarilabs — APPROVED** [review]
Thank you for incorporating required improvements and changes from https://github.com/kubeflow/model-registry/pull/74#pullrequestreview-2030284205 @isinyaaa 

/approve
/lgtm

**Pull request merged**